### PR TITLE
Make sankey nodes movable via drag and drop, closes #584

### LINKED ISSUE #584: Make sankey flow nodes draggable

### PR DESCRIPTION
**Pull request by fraenkelc**
I've also hit #391 and looked for a workaround that didn't involve fixes to the d3-sankey lib. A quick workaround here is to make the nodes movable via drag & drop. That way users can resolve overlaps themselves if / when they occur.

### CODE REVIEW

**Comment by dcbroad3** [comment]
FYI: #391 isn't due to the native d3-sankey library, that one seems to safely prevent the nodes from overlapping. It's actually caused by [d3-sankey-circular](https://github.com/tomshanley/d3-sankey-circular), a library I use to handle the circular loops in some recipe chains. The standard d3-sankey library throws an error if you include a circular loop, it isn't designed for those. However, d3-sankey-circular is a pretty drastic rewrite of d3-sankey and seems to include some bugs. 

Ideally, eventually, I hope to branch d3-sankey to include my own implementation of loops (as I believe Kirk McDonald's calculator did). d3-sankey itself is actually a really tiny library, though I'm not sure how difficult it would be to add loops; d3-sankey-circular is a much larger package. 

With all that being said, this seems like a good workaround, I'll look over the PR.

File: src/app/components/containers/flow-container/sankey/sankey.component.spec.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ import { SankeyComponent } from './sankey.component';
 class TestSankeyComponent {
   @ViewChild(SankeyComponent) child: SankeyComponent;
   sankeyData = Mocks.Sankey;
-  selectNode(data): void {}
+  selectNode(data): void { }
```
Comment (dcbroad3) [inline src/app/components/containers/flow-container/sankey/sankey.component.spec.ts:24]: Nitpick: Unnecessary space added here (possibly done automatically by a code style tool?)

Comment (fraenkelc) [reply src/app/components/containers/flow-container/sankey/sankey.component.spec.ts]: yeah, those where introduced when formatting the files with vscode.

Comment (dcbroad3) [reply src/app/components/containers/flow-container/sankey/sankey.component.spec.ts]: I figured it must be auto-formatting. Looks like the standard VS Code language settings add the space here, I use Prettier which removes them.

**Comment by fraenkelc** [comment]
this should fix the linting errors

**Comment by fraenkelc** [comment]
not sure on how to cover that particular branch - [the docs](https://observablehq.com/@d3/click-vs-drag?collection=@d3/d3-drag) list that particular check to be necessary but it does not seem to be triggered by the events I dispatch in the test :(

**Comment by dcbroad3** [comment]
> 
> 
> not sure on how to cover that particular branch - [the docs](https://observablehq.com/@d3/click-vs-drag?collection=@d3/d3-drag) list that particular check to be necessary but it does not seem to be triggered by the events I dispatch in the test :(

It looks like without it, when you drag the node it also gets selected. So I would prefer to keep that the way you wrote it. 
I tried to fiddle a bit with dispatching an event with preventDefault() called (so that defaultPrevented is true) but didn't have any luck yet. I'll try a bit more tonight. Absolute worst case, I think that function could be declared on the component so that you could pass in a custom event to it manually for testing. But, I think there must be a way to dispatch a custom event that would hit it during a test.

**Comment by dcbroad3** [comment]
I got a test working, but I'll just add it in a new branch. Thank you for the contribution! 😄

**Pull request merged**